ghcide/src/Development/IDE/Spans/Documentation.hs

```diff
@@ -73,32 +73,38 @@ getDocumentationsTryGhc env mod names = do
   res <- fun
   case res of
       Left _    -> return mempty
-      Right res -> fmap Map.fromList $ sequenceA $ unwrap <$> Map.toList res
+      Right res -> fmap Map.fromList $ sequenceA $ uncurry unwrap <$> Map.toList res
   where
     fun :: IO (Either [FileDiagnostic] (Map.Map Name (Either String (Maybe HsDocString, Map.Map Int HsDocString))))
     fun = catchSrcErrors (hsc_dflags env) "docs" $ getDocsBatch env mod names
 
-    unwrap :: (Name,  Either a (Maybe HsDocString, b)) -> IO (Name, SpanDoc)
-    unwrap (name, Right (Just docs, _)) = (name,) . SpanDocString docs <$> getUris name
-    unwrap (name, _)                    = (name,) . SpanDocText mempty <$> getUris name
-
```

Comment:
Since the right part of expression is the same & the left is close - creating the abstraction for the left part